parlai/core/opt.py

```diff
@@ -49,41 +49,37 @@ def __deepcopy__(self, memo):
         # track location of deepcopy
         loc = traceback.format_stack()[-3]
         self.deepcopies.append(loc)
-        # deepcopy the dict
-        memo = copy.deepcopy(dict(self))
-        # make into Opt object
-        memo = Opt(memo)
-        # deepcopy the history
-        memo.history = copy.deepcopy(self.history)
-        # deepcopy the deepcopy history
-        memo.deepcopies = copy.deepcopy(self.deepcopies)
+        # copy all our children
+        memo = Opt({k: copy.deepcopy(v) for k, v in self.items()})
```

Comment:
curious -- why is this faster than just deepcopying the dictionary?

Reply:
Bc you were making 2 shallow copies before the deepcopy.

Reply:
(The real savings is changing the history to be a shallow copy, the rest is fairly micro optimizations)